pandas/core/internals/blocks.py

```diff
@@ -294,6 +294,10 @@ def __setstate__(self, state):
         self.values = state[1]
         self.ndim = self.values.ndim
 
+    def __invert__(self):
+        new_values = ~self.values
+        return make_block(new_values, self.mgr_locs, ndim=self.ndim)
```

Comment:
Are we OK with defining such an op on the Block? (that doesn't seem like a pattern we have for other ops? It looks an easy solution though)

cc @jbrockmendel

Reply:
i agree - this is not a typical pattern for a block
why?

Reply:
What's the issue? The fact that it's a dunder op, or something else? Lots of other methods have a similar `return make_block(func(self.values))`

If the dunder op is a problem I can call it `.invert` I suppose.

Reply:
i think the alternative would be for `BlockManager.__invert__` to return `self._apply(operator.inv)`?

Reply:
Cool, didn't realize you could pass callables to BlockManager.apply.

Reply:
I would also prefer the BlockManager method to be `invert` instead of `__invert__`. In general when reading code and I see operations, I woudn't expect one of the operands to be a BlockManager.

Reply:
I don't think we'll even need BlockManager.invert. We'll just call `NDFrame._data.apply(operator.invert)`

Reply:
Just realized, this conversation was on `Block.__invert__`. Was it the problem, or was `BlockManager.__invert__` the problem?

Reply:
is this still needed? (as you use .apply above)

Reply:
Ohh `BlockManager.apply` calls `Block.apply(func)`, not `func(Block)`. So no, not needed.